org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/lookup/Scope.java

```diff
@@ -868,6 +868,9 @@ protected final MethodBinding computeCompatibleMethod(MethodBinding method, Type
 			if (CompilerOptions.tolerateIllegalAmbiguousVarargsInvocation && compilerOptions.complianceLevel < ClassFileConstants.JDK1_7)
 				tiebreakingVarargsMethods = false;
 		}
+
+		deduceSignaturePolymorphicAnnotation(method);
+
 		if ((parameterCompatibilityLevel(method, arguments, tiebreakingVarargsMethods)) > NOT_COMPATIBLE) {
 			if ((method.tagBits & TagBits.AnnotationPolymorphicSignature) != 0) {
```

Comment:
if `deduceSignaturePolymorphicAnnotation` stays in Scope, then invocation between the two `if`s would be my recommendation.